pkg/alertmanager/statefulset.go

```diff
@@ -659,6 +659,27 @@ func makeStatefulSetSpec(a *monitoringv1.Alertmanager, config Config, tlsAssetSe
 		amArgs = append(amArgs, fmt.Sprintf("--%s=%s", confArg.Name, confArg.Value))
 		volumes = append(volumes, configVol...)
 		amVolumeMounts = append(amVolumeMounts, configMount...)
+
+		for _, vol := range configVol {
```

Comment:
nit: This code seems exactly the same on all 3 components, part of me feels like we should be able to extract it somewhere else and simply call a function. WDYT? This obviously can also be done in a follow up

Reply:
Done in 5f34f9150056f261f8c52db6dfb799453d79d4e2, pending e2e tests confirmation 😅

Reply:
And done 🟢